### PR TITLE
Add proxy support

### PR DESCRIPTION
**Pull request by lopuhin**
Also reorganize tests a bit to make them shorter and use different MockServer instances.

The code is ready, but I'll need to debug tests on travis

### CODE REVIEW

**Comment by lopuhin** [comment]
Tests pass now, using a ``proxy.py`` proxy on travis to simplify setup.
@kmike can you review this one too? :)

**Comment by lopuhin** [comment]
One thing I'm not sure about is how to better expose it in the autologin-middleware. We can either pass ``HTTP_PROXY`` and ``HTTPS_PROXY`` directly, as we do with ``SPLASH_URL`` now, or prefix them with ``AUTOLOGIN_``.

File: tests/test_autologin.py [hunk 1]
```diff
@@ -30,39 +31,55 @@ def test_login_request():
...
+        cookies = self.al.auth_cookies_from_url(self.url, 'admin', 'secret')
         assert {c.name: c.value for c in cookies} == {'_auth': 'yes'}
 
+    def test_proxy(self):
+        assert 'localhost' not in self.url, 'proxy_bypass bypasses localhost'
+        cookies = self.al.auth_cookies_from_url(
+            self.url, 'admin', 'secret',
+            settings={'HTTP_PROXY': 'http://127.0.0.1:8123'},
+        )
+        assert {c.name: c.value for c in cookies} == {'_auth': 'yes'}
```
Comment (kmike) [inline tests/test_autologin.py]: this test checks that autologin works when HTTP_PROXY is passed, but not that proxy i used, right?

File: .travis.yml [hunk 2]
```diff
@@ -9,6 +9,8 @@ branches:
 
 install:
     - pip install -U tox codecov
+    - pip install proxy.py
+    - proxy.py --port 8123 &
```
Comment (kmike) [inline .travis.yml]: Is it here (and not in tox.ini) because proxy.py doesn't support Python 3?

File: tests/test_autologin.py (hunk 1 above)
Comment (lopuhin) [inline tests/test_autologin.py]: Yes, it would be better to check it, let me add that.

Comment (lopuhin) [inline tests/test_autologin.py]: Added the check - I set a custom header and check in the ``LoginCheckProxy``, also checking that it does not work without proxy beforehand. Thank for the suggestion!

File: .travis.yml (hunk 2 above)
Comment (lopuhin) [inline .travis.yml]: It works on python3, just did not think about adding it there. Now I start a twisted proxy from tests, so just ``tox`` works.
Now I like it much better, thanks!

**Pull request merged**